### PR TITLE
feat: update to ag-grid v27

### PR DESCRIPTION
**Pull request by areisle**
relates to #20

updates to ag-grid 27. this jumped 6 major versions so due to the api changes between 21 and 27, the `useGrid` and `ColumnConfiguration` had to be rewritten

### CODE REVIEW

**Review by elewis2 — DISMISSED** [review]

**Review by kttkjl — COMMENTED** [review]
Just the comment regarding the `.ref` 

thought on manual testing ... unrelated to this PR, when this happens, the application breaks, but it is fine on production 
ActiveFilters

I believe this is caused by the missing undefined/null check on [this line](https://github.com/bcgsc/pori_graphkb_client/blob/0f112a4c1e90effbc1f3aac3651b7f210decec85/src/views/DataView/components/ActiveFilters/index.js#L91)

File: src/components/QueryResultsTable/index.js [hunk 1]
```diff
@@ -41,20 +41,22 @@ JumpToRecord.propTypes = {
...
+
+    if (gridApi) {
       gridApi.sizeColumnsToFit();
 
       if (gridApi && data) {
         gridApi.setRowData(data);
       }
     }
-  }, [gridReady, gridApi, data]);
+  }, [grid, data]);
```
[the remark this answers is not in the document]
Comment (areisle) [reply src/components/QueryResultsTable/index.js:59]: - [ ] change to `grid.ref`

**Comment by areisle** [comment]
> though on manual testing ... unrelated to this PR, when this happens, the application breaks, but it is fine on production 

I created a new issue #39 to address this so that fix will be in a separate PR

File: src/components/QueryResultsTable/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (areisle) [reply src/components/QueryResultsTable/index.js]: [1a08da5](https://github.com/bcgsc/pori_graphkb_client/pull/38/commits/1a08da5fb1dfa1857b03beb8cee8f9847efd183e)

**Comment by creisle** [comment]
- Tested all pages with tables
- tested column configuration
- tested tsv export

**Review by creisle — APPROVED** [review]

**Pull request merged**